### ISSUE TITLE
Add a method in hook to return cursor for multiple parallel database operations

### ISSUE THREAD

**Issue by sunank200** [astronomer/airflow-provider-duckdb#6]
Currently, if multiple database operations are done on DuckDB parallelly, it doesn't work with the current provider by default. Reason for this is that in DuckDB a single connection is thread-safe but is locked for the duration of the queries, effectively serializing database access in this case.

As per [documentation](https://duckdb.org/docs/api/python/overview), the hook should also return the cursor instead. If you want to create a second connection to an existing database, you can use the cursor() method.

**Comment by pgzmnk** [comment]
Agreed. It might make sense to return the cursor on the catch statement of a try:catch block on the specific exception. 

```
sqlalchemy.exc.OperationalError: (duckdb.IOException) IO Error: Could not set lock on file "/tmp/db.duckdb": Resource temporarily unavailable
```
